util/tlgen.py

```diff
@@ -75,27 +75,16 @@ def main():
         log.error("Elaboration failed." + repr(xbar))
 
     # Generate
-    out_rtl, out_pkg, out_core = tlgen.generate(xbar)
+    results = tlgen.generate(xbar)
```

Comment:
I see that we do error handling in the call to `tlgen.generate` in topgen.py. I think we should probably do the same thing in both tlgen.py and topgen.py: either handle errors in both or neither.

(And if we decide to try to catch errors, we should probably also decide what we're catching, rather than using a bare except. I don't think we want to catch Ctrl-C's...)

Reply:
the `except` is to catch the exception while generating the output files with Mako template. I don't know which exception the Mako template generates, unfortunately. The exception is to show the exact location of the template file, anytime when the exception happens. Without that general `except` clause, the error message is quite unusable.

Reply:
Agreed about the bare except: I actually wrote something with Mako for the first time this morning (after writing that comment). Yes, you have to do it this way :-(